File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/model/utils/ReaderPreferences.kt
package dev.subfly.yabacore.model.utils

enum class ReaderTheme {
    SYSTEM,
    DARK,
    LIGHT,
    SEPIA,
}

enum class ReaderFontSize {
    SMALL,
    MEDIUM,
    LARGE,
}

enum class ReaderLineHeight {
    NORMAL,
    RELAXED,
}

data class ReaderPreferences(
    val theme: ReaderTheme = ReaderTheme.SYSTEM,
    val fontSize: ReaderFontSize = ReaderFontSize.MEDIUM,
    val lineHeight: ReaderLineHeight = ReaderLineHeight.NORMAL,
)

File: Darwin/YABACore/Model/Core/ReaderPreferences.swift
//
//  ReaderPreferences.swift
//  YABACore
//
//  Parity with Compose `ReaderTheme` / `ReaderFontSize` / `ReaderLineHeight` / `ReaderPreferences`.
//

import Foundation

public enum ReaderTheme: String, Sendable, CaseIterable {
    case system
    case dark
    case light
    case sepia
}

public enum ReaderFontSize: String, Sendable, CaseIterable {
    case small
    case medium
    case large
}

public enum ReaderLineHeight: String, Sendable, CaseIterable {
    case normal
    case relaxed
}

public struct ReaderPreferences: Sendable, Equatable {
    public var theme: ReaderTheme
    public var fontSize: ReaderFontSize
    public var lineHeight: ReaderLineHeight

    public init(
        theme: ReaderTheme = .system,
        fontSize: ReaderFontSize = .medium,
        lineHeight: ReaderLineHeight = .normal
    ) {
        self.theme = theme
        self.fontSize = fontSize
        self.lineHeight = lineHeight
    }
}
